### PR TITLE
[#159068543] Optimize Android binary with Proguard

### PR DESCRIPTION
**Pull request by Alex193a**
Work in progress.
By enabling Proguard (code obfuscation) the app does not start because RNMixpanel crashes (due to a null reference).
By enabling only the option to split the APK in multiple APKs depending on the architecture and removing the resources of the languages not supported has meant that the size of each APK has decreased by about 48-50%.
Each APK now has a size of 19-21MB against the approximately 40MB of the universal APK.

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: android/app/build.gradle [hunk 1]
```diff
@@ -121,6 +121,9 @@ android {
         versionCode System.getenv('CIRCLE_BUILD_NUM') ? System.getenv('CIRCLE_BUILD_NUM').toInteger() : 1
         versionName "1.1." + (System.getenv('CIRCLE_BUILD_NUM') ? System.getenv('CIRCLE_BUILD_NUM') : "1")
         multiDexEnabled true
+        // The resConfigs attribute will remove all not required localized resources while building the application,
+        // including the localized resources from libraries.
+        resConfigs 'en', 'it'
```
Comment (cloudify) [inline android/app/build.gradle]: do we have any localized resources? I believe we deal with localization with react-i18n - @fpersico ?

File: android/app/proguard-rules.pro [hunk 2]
```diff
@@ -15,3 +15,64 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
+
+# Disabling obfuscation is useful if you collect stack traces from production crashes
+# (unless you are using a system that supports de-obfuscate the stack traces).
+#-dontobfuscate
+
+# React Native
```
Comment (cloudify) [inline android/app/proguard-rules.pro]: if you copied the following rules from some reference, please add the link in the comment

File: android/app/build.gradle (hunk 1 above)
Comment (Alex193a) [reply android/app/build.gradle]: This concerns the localizations and resources of the **native libraries**

File: android/app/proguard-rules.pro (hunk 2 above)
Comment (Alex193a) [reply android/app/proguard-rules.pro]: up to line 64 is the official [Proguard.pro](https://github.com/facebook/react-native/blob/master/ReactAndroid/proguard-rules.pro) present in the Facebook repository, it must be added?

Comment (cloudify) [reply android/app/proguard-rules.pro]: yes please provide a link to the source, it could be useful for future reference, thanks

**Comment by fpersico** [comment]
@Alex193a is this "By enabling Proguard (code obfuscation) the app does not start because RNMixpanel crashes (due to a null reference)." still true or is already fixed?

**Comment by Alex193a** [comment]
@fpersico fixed in the commit [5a7d3de](https://github.com/teamdigitale/io-app/pull/980/commits/5a7d3debe734a2d9378357b38a751dfb44aa3b94).
It seems to works correctly now.

**Comment by cloudify** [comment]
@fpersico @Undermaken can you give a look before we merge?

**Comment by fpersico** [comment]
@cloudify The alpha lane must be changed before merging. Going to do it now.

**Comment by fpersico** [comment]
@cloudify All seem working correctly. The apkS get created correctly (https://circleci.com/gh/teamdigitale/io-app/10263). Can we merge?

**Pull request merged**